executor/cook/launcher.py

```diff
@@ -0,0 +1,181 @@
+"""
+A component for launching and monitoring processes for a task's commands.
+"""
+
+import os
+import time
+import logging
+import subprocess
+
+from collections import namedtuple
+
+class Command(
+        namedtuple('Command', ('value', 'name', 'async', 'guard', 'default'))
+):
+    """
+    An internal representation of a runnable command. Using a namedtuple helps catch
+    issues with data integrity.
+    """
+
+    def run(self, env):
+        if self.name is None:
+            stdout = open('stdout', 'w+')
+            stderr = open('stderr', 'w+')
+        else:
+            stdout = open('stdout.' + str(self.name), 'w+')
```

Comment:
With name being optional, I can't seem to find where a value gets set if one isn't specified.  Maybe I missed it.